### ISSUE TITLE
can we have a simple tutorial?

### ISSUE THREAD

**Issue by llogiq** [nagisa/llvm_build_utils.rs#3]
For example, it should be easy enough to build test::black_box with this, right?

**Comment by nagisa** [comment]
No, sadly you cannot implement equivalent for `black_box` without inline assembly. While there are many ways to implement a function which inhibits optimisations for a value, only the implementation with inline assembly is able to compile down to literally nothing.

**Comment by nagisa** [comment]
As for the tutorials, there’s number of examples in the [tests](https://github.com/nagisa/llvm_build_utils.rs/tree/master/tests) directory. That being said, I don’t recommend using this crate yet.
